hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/JpaTargetType.java

```diff
@@ -56,96 +59,53 @@ public class JpaTargetType extends AbstractJpaTypeEntity implements TargetType,
             @JoinColumn(name = "distribution_set_type", nullable = false, updatable = false, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_target_type_relation_ds_type"))})
     private Set<DistributionSetType> distributionSetTypes;
 
-    @OneToMany(targetEntity = JpaTarget.class, mappedBy = "targetType", fetch = FetchType.LAZY)
-    private Set<Target> targets;
-
-    /**
-     * Constructor
-     */
-    public JpaTargetType() {
-        // default public constructor for JPA
-    }
-
-    /**
-     * Constructor, legacy support where <code>key</code> is set to passed <code>name</code>.
-     *
-     * @deprecated will be removed
-     *
-     * @param name
-     *            of the type
-     * @param description
-     *            of the type
-     * @param colour
-     *            of the type
-     */
-    @Deprecated(forRemoval = true)
-    public JpaTargetType(final String name, final String description, final String colour) {
-        this(name, name, description, colour);
-    }
-
-    /**
-     * Constructor.
-     *
-     * @param key
-     *            of the type
-     * @param name
-     *            of the type
-     * @param description
-     *            of the type
-     * @param colour
-     *            of the type. It will be null by default
-     */
     public JpaTargetType(final String key, final String name, final String description, final String colour) {
         super(name, description, key, colour);
     }
 
     /**
-     * @param dsSetType
-     *          Distribution set type
+     * Adds a compatible distribution set type to this target type.
+     *
+     * @param dsSetType Distribution set type to add
      * @return Target type
      */
-    public void addCompatibleDistributionSetType(final DistributionSetType dsSetType) {
+    public JpaTargetType addCompatibleDistributionSetType(final DistributionSetType dsSetType) {
```

Comment:
not sure why this was changed :?
return type is not used

Reply:
why not? makes API fluent. It provides option to add fixed set of types fluently, e.g. in tests.
also make it in sync which removeDistributionSetType - also returns 'this' without being actually used.
Both could be made to return void in future

Reply:
i don't mind it, whenever i see return this i think of it as a builder/chain pattern. just noticed that with the change it was never used/referenced